site/pages/api/create-social-media-post.test.ts

```diff
@@ -0,0 +1,290 @@
+                ...previousCreateSocialMediaPostInformation,
+                socialMediaPostIndex: 0,
+            },
+            DEFAULT_ORG_ID,
+        );
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `${REVIEW_DISRUPTION_PAGE_PATH}/${defaultDisruptionId}`,
+        });
+    });
+
+    it("should redirect to /review-disruption when all required inputs are passed and an image", async () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: previousCreateSocialMediaPostInformation,
+            mockWriteHeadFn: writeHeadMock,
+        });
+        formParseSpy.mockResolvedValue({
+            fields: previousCreateSocialMediaPostInformation,
+
+            files: [
+                {
+                    size: 1000,
+                    filepath: "/testPath",
+                    newFilename: "testFile",
+                    mimetype: "image/jpg",
+                    mtime: null,
+                    originalFilename: "",
+                } as formidable.File,
+            ],
+        });
+        const buffer = Buffer.from("testFile", "base64");
+        await createSocialMediaPost(req, res);
+        readFileSpy.mockResolvedValue(buffer);
+        expect(s3Spy).toHaveBeenCalledTimes(1);
+        s3Spy.mockResolvedValue();
+        expect(upsertSocialMediaPostSpy).toHaveBeenCalledTimes(1);
+        expect(upsertSocialMediaPostSpy).toHaveBeenCalledWith(
+            {
+                ...previousCreateSocialMediaPostInformation,
+                socialMediaPostIndex: 0,
+                image: {
+                    filepath: "/testPath",
+                    key: "35bae327-4af0-4bbf-8bfa-2c085f214483/acde070d-8c4c-4f0d-9d8a-162843c10333/0.jpg",
+                    mimetype: "image/jpg",
+                    size: 1000,
+                },
+            },
+            DEFAULT_ORG_ID,
+        );
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `${REVIEW_DISRUPTION_PAGE_PATH}/${defaultDisruptionId}`,
+        });
+    });
+
+    it("should redirect to /review-disruption when all required inputs are passed and an image is size 0", async () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: previousCreateSocialMediaPostInformation,
+            mockWriteHeadFn: writeHeadMock,
+        });
+        formParseSpy.mockResolvedValue({
+            fields: previousCreateSocialMediaPostInformation,
+
+            files: [
+                {
+                    size: 0,
+                    filepath: "/testPath",
+                    newFilename: "testFile",
+                    mimetype: "application/octet-stream",
+                    mtime: null,
+                    originalFilename: "",
+                } as formidable.File,
+            ],
+        });
+        await createSocialMediaPost(req, res);
+
+        expect(upsertSocialMediaPostSpy).toHaveBeenCalledTimes(1);
+        expect(upsertSocialMediaPostSpy).toHaveBeenCalledWith(
```

Comment:
Might also be worth making sure that `putItem` isn't called here

Reply:
done